app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java

```diff
@@ -722,15 +723,21 @@ public boolean isStateBad() {
 
     // Save the file
     public boolean saveDocument(final boolean forceSaveEmpty) {
-        if (isSdStatusBad() || isStateBad()) {
+        final Activity activity = getActivity();
+        if (activity == null || isSdStatusBad() || isStateBad()) {
             errorClipText();
             return false;
         }
 
         // Document is written iff writeable && content has changed
         final CharSequence text = _hlEditor.getText();
         if (!_document.isContentSame(text)) {
-            if (_document.saveContent(getActivity(), text, _cu, forceSaveEmpty)) {
+            final int minLength = GsContextUtils.TEXTFILE_OVERWRITE_MIN_TEXT_LENGTH;
+            if (!forceSaveEmpty && text != null && text.length() < minLength) {
```

Comment:
Without this saving a small file (< 2 chars) would be treated as an error and the characters would be copied to the clipboard.